espressif/vscode-esp-idf-extension
# [VSC-1478] Disable cmds for Codespaces and redirect Flash monitor to Web extension

… web extension

## Description

Add `UIKind.Web` validation for Codespaces to disable commands not available in Browser or Codespaces and redirect Flash and monitor commands to ESP-IDF Web extension

## Type of change

- New feature (non-breaking change which adds functionality)

## Steps to test this pull request

Open [VSCode Extension codespace test](https://github.com/espressif/vscode-extension-codespace-test) in Codespaces. Uninstall the ESP-IDF extension that was pre installed. Load this PR vsix installer into the current Codespaces directory and install vsix from there to Codespaces (install from local doesn't work for me).

1. Click on `ESP-IDF: Flash your project` or Flash icon from status bar, the extension will call the ESP-IDF Web extension flash command. Same for `ESP-IDF: Monitor your device` to call web extension monitor command.
2. Click on any commands that should work in browser like `ESP-IDF: Build, flash and monitor`. You should receive a notification error message that command is not available in Web.
3. Observe results.

- Expected behaviour:

- Expected output:

## How has this been tested?

Manual testing using steps above.

**Test Configuration**:
* ESP-IDF Version: 5.3
* OS (Windows,Linux and macOS): Codespaces

## Checklist
